adding kubecon recap blog (#81)
metalbear-co/metalbear.co: content/en/blog/kubecon-2024-takeaways-what-we-learned-without-getting-our-heads-spun/index.md

---
title: "KubeCon 2024 Takeaways: What We Learned Without Getting Our Heads Spun"
description: "A summary of our collective impressions and insights from KubeConEU 2024"
lead: "A summary of our collective impressions and insights from KubeConEU 2024"
tags:
  - CNCF
  - KubeCon
  - cloud
  - application-testing  
  - platform-engineering 
date: 2024-04-03T0:00:00+00:00
lastmod: 2024-04-03T0:00:00+00:00
draft: false
weight: 50
contributors: ["Aviram Hassan", "Tal Zwick"]
---

Last week marked a huge milestone for us as we attended our first KubeCon! This was KubeCon EU which took place in Paris, city of lights and cloud orchestration. Below is a summary of our collective impressions and insights.

> # Co-Located Sessions and Surprising Highlights

Reviewer: ```suggestion
## Co-Located Sessions and Surprising Highlights
```

---

> title: "KubeCon 2024 Takeaways: What We Learned Without Getting Our Heads Spun"

Reviewer: ```suggestion
title: "KubeCon 2024 Takeaways"
```

The subtitle is a bit weird, I'd stay with just the first part

> description: "A summary of our collective impressions and insights from KubeCon Europe  2024"

Reviewer: ```suggestion
description: "A summary of our collective impressions and insights from KubeCon Europe 2024"
```

> lead: "A summary of our collective impressions and insights from KubeCon Europe  2024"

Reviewer: ```suggestion
lead: "A summary of our collective impressions and insights from KubeCon Europe 2024"
```

tags:
  - CNCF
  - KubeCon
  - cloud
  - application-testing  
  - platform-engineering 
date: 2024-04-03T0:00:00+00:00
lastmod: 2024-04-03T0:00:00+00:00
draft: false
weight: 50
images: ["thumbnail.png"]
contributors: ["Aviram Hassan", "Tal Zwick"]
---

> March marked a huge milestone for us as we attended our first KubeCon! This was KubeCon Europe, which took place in Paris, a city of lights and cloud orchestration. Below is a summary of our collective impressions and insights.

Reviewer: ```suggestion
Last month marked a huge milestone for us as we attended our first KubeCon! This was KubeCon Europe, which took place in Paris, city of lights and cloud orchestration. Below is a summary of our collective impressions and insights.
```

> ## Co-Located Sessions and Surprising Highlights

Reviewer: ```suggestion
## Co-Located Events and Surprising Highlights
```

> We arrived a day before KubeCon proper started so we could attend the co-located events. We ended up mostly going to AppDeveloperCon sessions, and it was surprisingly crowded. Our highlight from the first day was Daniel Bryant’s excellent talk “**Testing Cloud Apps: Mocks vs. Service Virtualization vs. Remocal Tools**” ([slides](https://speakerdeck.com/danielbryantuk/testing-cloud-apps-mocks-vs-service-virtualization-vs-remocal-tools), [video](https://www.youtube.com/watch?v=3tP31j_T6IE)). Daniel covered our favorite pain - being unable to work against a real environment and iterate with it as part of the development workflow. We were pleased to see a very friendly shoutout to mirrord and our to talk later in the convention! 

Reviewer: To keep things consistent, lets just have all talk titles link directly to the video. No need to link to the slides here.

Author: I agree

{{<figure src="first-image.png" alt="KubeCon 2024 Takeaways " height="100%" width="100%">}}

## The Main Event™

> The next day the main KubeCon event officially started, and it was PACKED. After two years of mostly working from our homes, we finally got to say hi to friends from all over the world who are part of our awesome community. Tal did his talk “[**When They Go High, We Go Low – Hooking Libc Calls to Debug Kubernetes Apps**](https://youtu.be/3OSQdiKTNU8?si=wP3IPbMM2g-GSBl-)”. There was a turnout of about 400 people, which was roughly 10 times what we expected - this was not lost on Tal. Feedback was incredibly positive though, and we lingered in the auditorium long after the talk to answer questions, discuss mirrord, and hand out some quality swag. 

Reviewer: The bold part doesn't render well - check the preview

Reviewer: (Same with the Rustvolution and Kubernetes Controllers link)

Our favorite talk, also on Thursday, was “[**Kubernetes Controllers in Rust: Fast, Safe, Sane**](https://kccnceu2024.sched.com/event/1YeOR/kubernetes-controllers-in-rust-fast-safe-sane-matei-david-buoyant)” by Matei David, also from Buoyant, where we found about Kubert, which might be really useful for some of our use cases! Kubert is built on top of kube-rs, a Rust crate for interacting with the Kubernetes API. It provides abstractions and batteries for building operators/controllers and can be compared to the Operator Framework.
We had a fun surprise on the last day of the conference when a team that uses mirrord came up to us with an informal bug report (the bug wasn’t the fun part). There’s nothing better for us than actually meeting the people who use the tool we built on a daily basis.

## Closing Thoughts

> To wrap things up: our first KubeCon was a blast. We had a chance to present the unique technical details of the implementation of mirrord, geek out about it with a great crowd, and meet our community face to face. We’re already waiting for KubeCon NA (at Salt Lake City, cloud-native capital of the western frontier!), and we might even do a booth this time. Looking forward to seeing everyone there!

Reviewer: ```suggestion
To wrap things up: our first KubeCon was a blast. We had a chance to present the unique technical details of the implementation of mirrord, geek out about it with a great crowd, and meet our community face to face. We’re already waiting for KubeCon North America (at Salt Lake City, cloud-native capital of the western frontier!), and we might even do a booth this time. Looking forward to seeing everyone there!
```
